SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -523,8 +527,19 @@ void USpatialReceiver::ReceiveActor(Worker_EntityId EntityId)
 			return;
 		}
 
+
 		EntityActor = TryGetOrCreateActor(UnrealMetadataComp, SpawnDataComp);
 
+		// RemoveActor immediately if we've received the tombstone component.
+		// We must first Resolve the EntityId to the Actor in order for RemoveActor to succeed.
+		if (NetDriver->StaticComponentView->GetComponentData<Tombstone>(EntityId) != nullptr)
+		{
+			UE_LOG(LogSpatialReceiver, Verbose, TEXT("The received actor with entity id %lld was tombstoned. The actor will not be spawned."), EntityId);
+			PackageMap->ResolveEntityActor(EntityActor, EntityId);
```

Comment:
Why do we need to resolve the actor in the package map?

Reply:
Without the resolve, the call on the next line - RemoveActor - fails immediately internally because this returns a nullptr:

TWeakObjectPtr<UObject> WeakActor = PackageMap->GetObjectFromEntityId(EntityId);